### PR TITLE
fixed exception handling in secretsmanager

### PR DESCRIPTION
**Pull request by macnev2013**
## Motivation
This PR addresses the issue described in https://github.com/localstack/localstack/issues/10621.

## Changes
Updated moto exception handling in LocalStack.

## Testing

- Added aws validated test cases.

### CODE REVIEW

File: tests/aws/services/secretsmanager/test_secretsmanager.py [hunk 1]
```diff
@@ -2337,6 +2337,33 @@ def test_secret_tags(self, aws_client, create_secret, sm_snapshot, cleanups):
...
+        )
+        version_id = response["VersionId"]
+
+        sm_snapshot.add_transformers_list(
+            sm_snapshot.transform.secretsmanager_secret_id_arn(response, 0)
+        )
+
+        secret_arn = response["ARN"]
+
+        with pytest.raises(Exception) as ex:
```
Comment (viren-nadkarni) [inline tests/aws/services/secretsmanager/test_secretsmanager.py:2355]:
```suggestion
        with pytest.raises(ClientError) as ex:
```

File: tests/aws/services/secretsmanager/test_secretsmanager.py [hunk 2]
```diff
@@ -2337,6 +2337,33 @@ def test_secret_tags(self, aws_client, create_secret, sm_snapshot, cleanups):
...
+            aws_client.secretsmanager.get_secret_value(
+                SecretId=secret_arn, VersionStage="AWSPREVIOUS"
+            )
+        sm_snapshot.match("error_get_secret_value_non_existing", ex.value.response)
+
+        with pytest.raises(Exception) as exc:
```
Comment (viren-nadkarni) [inline tests/aws/services/secretsmanager/test_secretsmanager.py:2361]:
```suggestion
        with pytest.raises(ClientError) as exc:
```

File: localstack/services/secretsmanager/provider.py [hunk 3]
```diff
@@ -21,7 +21,12 @@
 from moto.secretsmanager.exceptions import (
     SecretHasNoValueException as MotoSecretHasNoValueException,
 )
-from moto.secretsmanager.exceptions import SecretNotFoundException as MotoSecretNotFoundException
+from moto.secretsmanager.exceptions import (
+    SecretNotFoundException as MotoSecretNotFoundException,
+)
+from moto.secretsmanager.exceptions import (
+    SecretStageVersionMismatchException as MotoSecretStageVersionMismatchException,
+)
```
Comment (viren-nadkarni) [inline localstack/services/secretsmanager/provider.py:29]: (non-blocking) Consider importing `moto.secretsmanager.exceptions` and using exceptions by attribute access

**Review by viren-nadkarni — CHANGES_REQUESTED** [review]

File: localstack/services/secretsmanager/provider.py [hunk 4]
```diff
@@ -6,22 +6,10 @@
 import time
 from typing import Final, Optional, Union
 
+import moto.secretsmanager.exceptions as MotoException
```
Comment (viren-nadkarni) [inline localstack/services/secretsmanager/provider.py:9]: Please don't deviate from Python naming conventions:

https://peps.python.org/pep-0008/#package-and-module-names

Modules must use lowercase (with underscores for separators)

**Review by viren-nadkarni — CHANGES_REQUESTED** [review]

File: localstack/services/secretsmanager/provider.py (hunk 4 above)
Comment (macnev2013) [reply localstack/services/secretsmanager/provider.py]: Updated. thanks for the suggestion.

**Review by viren-nadkarni — APPROVED** [review]

**Pull request merged**